packages/server-wallet/src/protocols/channel-opener.ts

```diff
@@ -38,34 +38,61 @@ export class ChannelOpener {
       if (!channel) {
         throw new Error(`ChannelOpener can't find channel with id ${channelToLock}`);
       }
+      // always queue the channel if we've potentially touched it
+      response.queueChannel(channel);
 
       // if we haven't signed the pre-fund, sign it
       if (!channel.prefundSigned) {
         await this.signPrefundSetup(channel, response, tx);
       }
 
-      // if we don't have a complete preFundSetup, we're done for now
+      // We might be waitingFor theirPreFundState
+      let waitingFor = 'theirPreFundState';
+
+      // if we don't have a complete preFundSetup, we are still waitingFor theirPreFundState and cannot progress
       if (!channel.preFundComplete) {
-        response.queueChannel(channel); // always queue the channel if we've potentially touched it
+        await updateWaitingFor();
         return;
```

Comment:
It seems as if `updateWaitingFor` is always followed by `return`. It also seems as if that's an important property. Why not `return updateWaitingFor();`?

Reply:
Yeah that's a good idea. The follow-on PR goes down a very similar route, in fact.